src/latency.h

```diff
@@ -91,34 +91,39 @@ void latencyAddSample(const char *event, mstime_t latency);
 #define latencyRemoveNestedEvent(event_var,nested_var) \
     event_var += nested_var;
 
-struct durationStats {
+typedef struct durationStats {
     unsigned long long cnt;
     long long sum;
     long long max;
-};
+} durationStats;
+
+#define EL_DURATION_TYPE_NUM 6
 
-void durationAddSample(const char *event, ustime_t duration, int add_to_latency);
+typedef enum {
+    EL_DURATION_TYPE_EL = 0,
+    EL_DURATION_TYPE_CMD,
+    EL_DURATION_TYPE_IO_READ,
+    EL_DURATION_TYPE_IO_WRITE,
+    EL_DURATION_TYPE_AOF,
+    EL_DURATION_TYPE_CRON
+} DurationType;
+
+void durationAddSample(int type, ustime_t duration);
 
 /* Record the start time of an event. In order to avoid performance implication due to 
  * querying the clock using a system call every time we measure a duration, we use a 
  * monotonic clock, when we are sure its cost is very low, and fall back to non-monotonic 
  * call otherwise. */
-#define durationStartMonitor(var)                   \
-    if (monotonicGetType() == MONOTONIC_CLOCK_HW) { \
-        var = getMonotonicUs();                     \
-    } else {                                        \
-        var = ustime();                             \
-    }
-
-/* End monitoring an event, compute the difference with the current time to check the 
+#define durationStartMonitor(var) \
+    do {                          \
+        var = getMonotonicUs();   \
+    } while (0)
```

Comment:
i'm not sure we need the macros.. it seems simple enough if the code would just explicitly call getMonotonicUs